src/node/blockstorage.cpp

```diff
@@ -1067,6 +1067,11 @@ bool BlockManager::ReadBlockFromDisk(CBlock& block, const CBlockIndex& index) co
 bool BlockManager::ReadRawBlockFromDisk(std::vector<uint8_t>& block, const FlatFilePos& pos) const
 {
     FlatFilePos hpos = pos;
+    // If nPos is less than 8 the pos is null and we don't have the block data
+    // Return early to prevent undefined behavior of unsigned int underflow
+    if (hpos.nPos < 8) {
+        return error("%s: OpenBlockFile failed for %s", __func__, pos.ToString());
```

Comment:
In 3f086a9353191a5df03aa595bdbb42c2195a23d2 "blockstorage: check nPos in ReadRawBlockFromDisk before seeking back"

Silent merge conflict here:

```
../../../src/node/blockstorage.cpp: In member function ‘bool node::BlockManager::ReadRawBlockFromDisk(std::vector<unsigned char>&, const FlatFilePos&) const’:
../../../src/node/blockstorage.cpp:1094:16: error: ‘error’ was not declared in this scope; did you mean ‘perror’?
 1094 |         return error("%s: OpenBlockFile failed for %s", __func__, pos.ToString());
      |                ^~~~~
      |                perror
```

`error()` was removed in #29236

Reply:
@achow101 resolved, thanks.